docs/DeveloperGuide.md

```diff
@@ -300,6 +304,97 @@ if command is valid. Command is invalid if user input is empty, or if user enter
+1. The new `Alias` and `AliasCommand` is created, and executed.
+1. The alias is added to both the parser, and the model (so that it can be saved in the `UserPrefs`)
+    * If the alias word is already present in the parser, the command word it is mapped to is replaced with the new command word instead.
+    * If the alias word and the command word are identical, the existing alias is removed instead.
+    
+If the command word of any future user input matches the alias, the first word of the user input will be replaced by the command word of the alias.
+The activity diagram of how a command is parsed can be found in the [Logic section of this guide](#logic-component).
+
+#### Design Considerations
+
+**Aspect 1: How aliases are stored and how they parse**
+
+There were two ideas on how this could be done:
+1. Store the command's `Parser` in each alias. The parser attribute of the alias can then be retrieved and `Parser::execute` can be called when parsing.
+    * Pros:
+        * Very elegant and streamlined parsing. Almost no changes need to be made to the logic of `AddressBookParser`.
+    * Cons:
+        * It is not easy to obtain the type of `Parser` simply through the command word provided by the user given the current implementation of the project. You'd need to iterate through all possible `Commands`, and match their command word with the provided word. This approach would not be very extensible if more commands are to be added in the future. 
+        * It is also not easy to save/load the alias to/from the `UserPrefs`. The best approach is likely to just save the CommandWord, then generate the Parser when loading. However this poses a similar problem as the previous point, which is that it is not easy to generate the `Parser` just from the command word.
+1. Store the command word in each alias. Replace the command word of the user input if the user inputs an aliased command, and parse it again.
+    * Pros:
+        * Very easy to store. Each alias is essentially just two strings.
+    * Cons: 
+        * Have to be careful with the implementation to prevent any edge cases where the user can define aliases to create an infinite loop.
+    
+We decided to go with the easier implementation of storing each alias as two strings. However, there were still more aspects to be considered.
+
+**Aspect 2: How to handle aliases of aliases**
+
+That is, what happens when the user does `alias -c <existing_alias> -as <new_alias>`? There are two choices for this: 
+1. The new alias maps to the existing alias word directly.
+    * The command word stored in the new `Alias` would be the alias word of the existing `Alias`.
+    * Assuming that an alias `as` already exists for `addstudent`, the following commands have the following effects:
+        * `alias -c as -as as2` would make `as2` map to `addstudent` via `as`.
```

Comment:
```suggestion
        * `alias -c as -as as2` would make `as2` map to `as`, which will be then mapped to `addstudent` via `as` when used.
```
Would this better convey the meaning? I think the "make `as2` map to `addstudent`" may trip up the reader